collaborators.py

```diff
@@ -0,0 +1,22 @@
+import util
+
+def update(repo, users):
+  collaborators = [c['login'] for c in util.request(_collaborators_url(repo)).json()]
+  _remove_non_players(repo, users, collaborators)
+  _add_new_players(repo, users, collaborators)
+
+def _remove_non_players(repo, users, collaborators):
+  for collaborator in collaborators:
+    if not collaborator in users:
+      print('%s is a collaborator, but not a player. Removing their status as a collaborator' % collaborator)
+      util.request_delete('%s/%s' % (_collaborators_url(repo), collaborator))
+
+
+def _add_new_players(repo, users, collaborators):
+  for user in users:
+    if not user in collaborators:
+      print('%s is a player, but not a collaborator. Adding them as a collaborator' % user)
+      util.request_put('%s/%s' % (_collaborators_url(repo), user))
+
+def _collaborators_url(repo):
+  return 'https://www.jefftk.com/nomic-github/repos/%s/collaborators' % repo
```

Comment:
hmm, I should have a whitelist of what kind of requests to proxy

Reply:
I assume this won't actually work until you set that up? Or is the issue that you're currently letting anything through?

Reply:
Before 528a406 it was letting through everything under https://api.github.com/repos/jeffkaufman/nomic but now it's restricted to https://api.github.com/repos/jeffkaufman/nomic/pulls  Still, this would allow someone to PUT to https://api.github.com/repos/jeffkaufman/nomic/pulls/N/merge and since I'm an admin I think they might be able to merge without Travis passing.  So in 83f90b7 I limited it to only GET requests.

If we decide to poke a hole through for `collaborators` we can, but we should be careful about it.  For example, I don't know what would happen if you removed me!

Reply:
I'm guessing it would fail, though I'm not inclined to test it. We could have it skip over you, or whoever is the repo owner. That doesn't address the broader proxy issue, though, just the code in this PR.

Reply:
Right, but since anyone can poke the proxy we need the proxy to do validation.

Reply:
The easiest thing to do might be to stand up your own nginx?  Then I don't have to figure out how to do the validation...

Reply:
Maybe I'm misunderstanding something, but how would that give me permissions? I don't have your GitHub credentials to use. Are you saying that I could hit the proxy without you needing to update the whitelist?

Reply:
The idea is you'd use your own github credentials and play around with the `tnelling/nomic` repo.  But I'm not sure there's a good way to open this up?  Letting anyone add/remove collaborators doesn't seem great, and that's what the proxy would have to do.

Reply:
Ah, of course. Might generate some annoying email traffic, but would work otherwise.

Could the proxy only let traffic through to `collaborators` from Travis (`travis-ci.com/jeffkaufman/nomic`, even)? And could we consider requests to `collaborators` to be out-of-bounds?

Reply:
If there was a way to restrict it to only builds run on master, I'd suggest that, but I can't see how the proxy could possibly tell the difference.